accepted/2024/support-nuget-authentication-plugins-dotnet-tools.md

```diff
@@ -0,0 +1,500 @@
+However, given the limited number of plugin implementations currently available, this is not a problem that needs to be solved at this point in time, in my understanding.
+
+## Prior Art
+
+- The AWS team has developed a [.NET tool](https://docs.aws.amazon.com/codeartifact/latest/ug/nuget-cli.html#nuget-configure-cli) to simplify authentication with their private NuGet feeds.
+In their current implementation, they've added a subcommand, `codeartifact-creds install`.
+This command copies the credential provider to the NuGet plugins folder.
+They also provide an `uninstall` subcommand to remove the files from the NuGet plugin folders.
+However, due to limitations in the NuGet Client tooling, they've had to maintain plugins for both .NET Framework and .NET Core tooling. Additionally, they've had to provide commands to install and uninstall the NuGet plugins.
+
+## Unresolved Questions
+
+- The issue regarding workload installation on Mac and Linux, as well as NuGet credential providers, is discussed in [GitHub Issue #35912](https://github.com/dotnet/sdk/issues/35912#issuecomment-1759774310).
+When executing .NET SDK workload commands under sudo, the HOME directory path is modified, preventing the .NET SDK from writing files owned by root to the user's regular HOME directory.
+This alteration can cause problems when running standard commands without sudo and also interferes with NuGet credential providers.
+We have proposed a [workaround](https://github.com/dotnet/sdk/issues/35912#issuecomment-2004522180) for this issue.
+
+## Future Possibilities
+
+### Managing NuGet plugins per repository using .NET Local Tools
+
+- A potential future possibility is mentioned under the `Specify the authentication plugin in NuGet.Config file` section.
+In addition to that, if we ever plan to provide users with an option to manage their plugins per repository instead of loading all the available plugins, we can consider using [.NET Local tools](https://learn.microsoft.com/dotnet/core/tools/local-tools-how-to-use).
+- The advantage of local tools is that the .NET CLI uses manifest files to keep track of tools that are installed locally to a directory.
+When the manifest file is saved in the root directory of a source code repository, a contributor can clone the repository and invoke a single .NET CLI command such as [`dotnet tool restore`](https://learn.microsoft.com/dotnet/core/tools/dotnet-tool-restore) to install all of the tools listed in the manifest file.
+- Having NuGet plugins under the repository folder eliminates the need for NuGet to load plugins from the user profile.
+
+### Support for NuGet plugins developed in non-.NET languages
+
+- The NuGet Client plugin code needs to be updated to utilize a standard RPC mechanism, such as StreamJsonRpc, in order to support NuGet plugins developed in languages other than .NET.
```

Comment:
It does not need to be, it just makes it margionally easier to implement.

FWIW, I feel like this section is out of scope for this PR. By scanning PATH for executables with a certain naming pattern, we are already enabling plugins developed in non-.NET languages. Therefore there's not else we need to do in the future to enable it. Making it easier via using a commonly used RPC mechanism is completely orthogonal to the goals this PR is about (easy for customers to install)

Reply:
Agreed with your feedback. Removed this section given we added support for `.bat` file.